### PR TITLE
Better error message for a missing setting.yaml file

### LINKED ISSUE #2467: Display better message when not finding default epinio `settings.yaml` file
## Issue
Currently if default epinio settings.yaml file is not found the message displayed is:

```
❌  error retrieving Epinio environment information: making the request: Get "/api/v1/info": unsupported protocol scheme ""
```

## Request

It would be nice to say something as  `"settings.yaml" file not found in /home/mmartin/.config/epinio/` or similar

## To reproduce:
- remove `/home/mmartin/.config/epinio/settings.yaml ` from your respective path
- run `epinio info`

### PR DESCRIPTION
**Pull request by andreas-kupries**
Fix #2467

### CODE REVIEW

**Review by enrichman — CHANGES_REQUESTED** [review]
This is not checking for the existance of the settings file but for the API url to be set. Still fine to have it, but this will prevent the user from understanding if it's using a non-existing file or just a settings file with the url not set.

We already have this check [here](https://github.com/epinio/epinio/blob/main/internal/cli/settings/settings.go#L99-L102), but it seems it doesn't work: 

```go
settingsExists, err := fileExists(file)
if err != nil {
	return nil, errors.Wrapf(err, "filesystem error")
}
```

Maybe [this](https://github.com/epinio/epinio/blob/main/internal/cli/settings/settings.go#L84) is preventing the check?

```go
v.SetConfigFile(file)
```

**Comment by andreas-kupries** [comment]
> This is not checking for the existence of the settings file but for the API url to be set.

Yes, intentionally.

> ... will prevent the user from understanding if it's using a non-existing file or just a settings file with the url not set.

I do not see the user caring about this distinction.
The result of either is the same, failure to talk to the E API server.
The solution is the same too, logging into the desired E API endpoint.

I believe that the solution is good as-is.

**Comment by enrichman** [comment]
Actually I opened the issue because one of the bug that we have found was probably not catched because of this.
Also if in the code the check exists but is not triggered that means that there is a flaw in the logic and a bug that we should fix. Basically why the `fileExists` doesn't work?

There are also different scenarios where the information is useful. I'm thinking about the `epinio` cli running on a CI, and having this error for a typo in the epinio settings location. It will be very hard or difficult to understand why it's not working, while having a "settings not found" will be much easier to understand.

**Comment by andreas-kupries** [comment]
> We already have this check [here](https://github.com/epinio/epinio/blob/main/internal/cli/settings/settings.go#L99-L102), but it seems it doesn't work:
> Maybe [this](https://github.com/epinio/epinio/blob/main/internal/cli/settings/settings.go#L84) is preventing the check?

No. This tells viper where to look for the file. This is independent of the other check

I rather suspect that the error from `LoadFrom` is simply ignored by some caller.

Ok, will have another look and see if a working file check can be added.

**Comment by enrichman** [comment]
Just a small comment, but it's ok. I can approve it right after the small change (if agreed), then CI should be fine!

File: internal/cli/usercmd/login.go [hunk 1]
```diff
@@ -316,6 +316,11 @@ func updateSettings(address, username, password, serverCertificate string) (*set
 }
 
 func verifyCredentials(ctx context.Context, epinioSettings *settings.Settings, customHeaders map[string]string) error {
+	// Ensure that the settings have a location to get passed the client's check for it.
+	// Here it is ok to not have an actual location and file, because we are logging in, and at
+	// the end of the operation the relevant file will be created. Only having an API matters,
+	// and we do.
+	epinioSettings.Location = "fake"
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply internal/cli/usercmd/login.go:323]: Ok

[the remark this answers is not in the document]
Comment (andreas-kupries) [reply internal/cli/usercmd/login.go:323]: Wwll, not quite. The helper used by the caller ... Looks good enough.

[the remark this answers is not in the document]
Comment (andreas-kupries) [reply internal/cli/usercmd/login.go:323]: Looking a bit more I believe that your request will break OIDC mode of `login`.
That has no `updateSettings`, just `verifyCredentials`.
The config is directly set up, would need a copy of the Location code.
Having it in `verify...` keeps it DRY.

[the remark this answers is not in the document]
Comment (andreas-kupries) [reply internal/cli/usercmd/login.go:323]: I decline to make the change.

**Review by enrichman — APPROVED** [review]

**Pull request merged**